setup.py

```diff
@@ -13,6 +14,31 @@
 except ImportError:
     readme = desc
 
+
+def setup(*args, **kwargs):
+    """This wrapper exists to force the option --build-type=ReleaseWithDocs.
+
+    Neither Release nor RelWithDebInfo will work, due to hard-coded options in
+    scikit-build's UseCython.cmake that disable docstrings. The choice of
+    ReleaseWithDocs is arbitrary, as a string that won't overlap with any build
+    type handled in UseCython.cmake. See this issue for details:
+    https://github.com/scikit-build/scikit-build/issues/518
+    """
+    BUILD_TYPE = '--build-type=ReleaseWithDocs'
+    for index, arg in enumerate(sys.argv):
+        if arg == '--':
+            # Insert at the end of the options that go to scikit-build
+            break
+        elif arg.startswith('--build-type') and arg != BUILD_TYPE:
+            raise RuntimeError(f'Specifying --build-type is not allowed. '
+                               f'freud requires {BUILD_TYPE}')
```

Comment:
I don't see any reason to do this. Users should be free to override build types if they wish, that's "here be dragons" territory. Enabling debug flags in particular could be of use in certain cases.

Reply:
Agreed. I will remove this failure and instead just rely on a default value.